compose/ui/ui/src/uikitInstrumentedTest/kotlin/android/compose/ui/test/UIKitInstrumentedTest.kt

```diff
@@ -0,0 +1,200 @@
+        val keyboardHeight = screenSize.height - height
+        val keyboardDuration = if (animated) 0.25 else 0.0
+        return mapOf(
+            UIKeyboardAnimationDurationUserInfoKey to NSNumber(double = keyboardDuration),
+            UIKeyboardFrameEndUserInfoKey to NSValue.valueWithCGRect(
+                CGRectMake(
+                    x = 0.0,
+                    y = keyboardHeight.value.toDouble(),
+                    width = screenSize.width.value.toDouble(),
+                    height = keyboardHeight.value.toDouble()
+                )
+            ),
+            UIKeyboardAnimationCurveUserInfoKey to NSNumber(unsignedInteger = 0uL)
+        )
+    }
+
+    fun showKeyboard(height: Dp = 216.dp, animated: Boolean = true) {
+        NSNotificationCenter.defaultCenter().postNotification(
+            NSNotification(
+                name = UIKeyboardWillShowNotification,
+                `object` = null,
+                userInfo = keyboardUserInfo(height, animated)
+            )
+        )
+        keyboardHeight = height
+        resizeKeyboard(height, animated)
+    }
+
+    fun resizeKeyboard(newHeight: Dp, animated: Boolean = true) {
+        NSNotificationCenter.defaultCenter().postNotification(
+            NSNotification(
+                name = UIKeyboardWillChangeFrameNotification,
+                `object` = null,
+                userInfo = keyboardUserInfo(newHeight, animated)
+            )
+        )
+        keyboardHeight = newHeight
+    }
+
+    fun hideKeyboard(animated: Boolean = true) {
+        NSNotificationCenter.defaultCenter().postNotification(
+            NSNotification(
+                name = UIKeyboardWillChangeFrameNotification,
+                `object` = null,
+                userInfo = keyboardUserInfo(0.dp, animated)
+            )
+        )
+        keyboardHeight = 0.dp
+    }
+
+    private val isIdle: Boolean get() {
+        val hadSnapshotChanges = Snapshot.current.hasPendingChanges()
+        val hasPendingRecomposer = Recomposer.runningRecomposers.value.any {
+            it.hasPendingWork
+        }
+        val isApplyObserverNotificationPending = Snapshot.isApplyObserverNotificationPending
+
+        return !hadSnapshotChanges && !hasPendingRecomposer && !isApplyObserverNotificationPending
+    }
+
+    @OptIn(ExperimentalNativeApi::class)
+    fun waitForIdle(deadline: Duration = 5.seconds): Boolean {
+        val defaultIdleScore = 10
+        var idleScore = defaultIdleScore
+        val timeoutAchieved = waitForExpectation(deadline = deadline) {
+            if (isIdle) {
+                idleScore -= 1
+            } else {
+                idleScore = defaultIdleScore
+            }
+            idleScore == 0
+        }
+        assert(timeoutAchieved) { "waitForIdle reached $deadline deadline" }
+        return timeoutAchieved
+    }
+
+    fun waitForExpectation(deadline: Duration = 5.seconds, expectation: () -> Boolean): Boolean {
```

Comment:
I'd recomment (re)using the existing API name/form: 
```
override fun waitUntil(
    conditionDescription: String?,
    timeoutMillis: Long,
    condition: () -> Boolean
)
```

Reply:
Sure, dont